eth/protocols/snap/sync.go

```diff
@@ -1553,6 +1553,12 @@ func (s *Syncer) processAccountResponse(res *accountResponse) {
 	// Ensure that the response doesn't overflow into the subsequent task
 	last := res.task.Last.Big()
 	for i, hash := range res.hashes {
+		// Mark the range complete if the last is already included.
+		// Keep iteration to delete the extra states if exists.
+		if hash.Big().Cmp(last) == 0 {
```

Comment:
Nitpick, `hash.Big().Cmp(last)` will almost never be 0, so we'll almost always redo the exact same calculation below. Might as well be a bit smarter?

```go
cmp := hash.Big().Cmp(last) 
if cmp == 0 {
...
}
if cmp > 0 {
...
}
```

?